### PR TITLE
Migrate bicep example: 101/nat-gateway-vnet

### PR DESCRIPTION
**Pull request by StephenWeatherford**
Added bicep support to this sample by integrating bicep.main from the example in https://github.com/Azure/bicep/tree/main/docs/examples/101/nat-gateway-vnet

The corresponding PR for the modified bicep example is here: https://github.com/Azure/bicep/pull/3236

### CODE REVIEW

File: quickstarts/microsoft.network/nat-gateway-vnet/azuredeploy.json [hunk 1]
```diff
@@ -123,24 +128,27 @@
         ],
         "enableDdosProtection": false,
         "enableVmProtection": false
-      }
+      },
+      "dependsOn": [
+        "[resourceId('Microsoft.Network/natGateways', parameters('natGatewayName'))]"
+      ]
     },
     {
```
Comment (StephenWeatherford) [inline quickstarts/microsoft.network/nat-gateway-vnet/azuredeploy.json:136]: It seems to me this subnet resource is completely redundant, as it's declared under the vnet, and should be removed.  Do you agree?

File: quickstarts/microsoft.network/nat-gateway-vnet/main.bicep [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+resource natGateway 'Microsoft.Network/natGateways@2020-06-01' = {
+  name: natGatewayName
+  location: location
+  sku: {
+    name: 'Standard'
+  }
+  properties: {
+    idleTimeoutInMinutes: 4
+    publicIpAddresses: !empty(publicIpDns) ? publicIpAddresses : null
```
Comment (StephenWeatherford) [inline quickstarts/microsoft.network/nat-gateway-vnet/main.bicep:53]: I'm assuming null is the same as json('null'), which is what this originally was.

**Comment by StephenWeatherford** [comment]
FYI @MCKLMT, @asudbring

File: quickstarts/microsoft.network/nat-gateway-vnet/main.bicep (hunk 2 above)
Comment (bmoore-msft) [reply quickstarts/microsoft.network/nat-gateway-vnet/main.bicep:53]: afaiki

**Pull request merged**